### ISSUE TITLE
Error while running the descheduler code on Windows 10

### ISSUE THREAD

**Issue by rajesh-krishnan-tech** [kubernetes-sigs/descheduler#318]
Hi, 
  I have gone through the contributor guide and followed the steps. 

1. Install go
2. Install make

Build descheduler.

cd $GOPATH/src/sigs.k8s.io
git clone https://github.com/kubernetes-sigs/descheduler.git
cd descheduler
make

Getting this error. Not sure where to check for logs. Please guide.

C:\Go\src\sigs.k8s.io\descheduler>make
process_begin: CreateProcess(NULL, which golangci-lint, ...) failed.

Has anyone run the code on Windows?

**Comment by seanmalloy** [comment]
I have not tried compiling or running the code on Windows. Looking at the Makefile there are several lines that assume a Unix like OS.

I can try to hack together a Makefile that might work on Windows.

**Comment by seanmalloy** [comment]
@rajesh-krishnan-tech I added a `Makefile` to [this gist](https://gist.github.com/seanmalloy/ec2e38c90f7feb8bb28bdaa6c4924268). This might work on Windows, but I have no way to test it as I don't have access to a Windows machine. This is just the minimum to hopefully get `make build` working for you.

If this does not work then you might want to try the [Windows 10 Linux Subsystem](https://docs.microsoft.com/en-us/windows/wsl/faq#:~:text=The%20Windows%20Subsystem%20for%20Linux,desktop%20and%20modern%20store%20apps.).

**Comment by seanmalloy** [comment]
/kind cleanup
/good-first-issue

I know I won't be able work on this, and I'm guessing the maintainers of this project will probably also not be able to work on this. The `Makefile` would need to be updated to work on Windows and some sort of Windows compatible scripts would need to be added in the `hack` directory.

I'd be happy to review a pull request if someone want to try and get this working.

**Comment by rajesh-krishnan-tech** [comment]
@seanmalloy - Tried running with the Makefile [here](https://gist.github.com/seanmalloy/ec2e38c90f7feb8bb28bdaa6c4924268).

Getting the below error
C:\Go\src\sigs.k8s.io\descheduler>make
CGO_ENABLED=0 go build -ldflags "-X sigs.k8s.io/descheduler/cmd/descheduler/app.version=v0.10.0-65-g25336da70 -X sigs.k8s.io/descheduler/cmd/descheduler/app.buildDate=hardcodedforwindows -X sigs.k8s.io/descheduler/cmd/descheduler/app.gitCommit=25336da70804caf39105934ed1baaa759ae04c11" -o _output/bin/descheduler sigs.k8s.io/descheduler/cmd/descheduler
'CGO_ENABLED' is not recognized as an internal or external command,
operable program or batch file.
make: *** [build] Error 1

**Comment by seanmalloy** [comment]
@rajesh-krishnan-tech I made one more tweak to the `Makefile` [here](https://gist.github.com/seanmalloy/ec2e38c90f7feb8bb28bdaa6c4924268). Maybe it will work for you now.

I honestly think your best chance of making it work on Windows 10 is to use the  Windows 10 Linux Subsystem.

**Comment by rajesh-krishnan-tech** [comment]
@seanmalloy  - It did succeed without any error.

Do we update the documentation for the benefit of the other Windows users?

**Comment by seanmalloy** [comment]
@rajesh-krishnan-tech I'm glad we were able to get `make build` working for you.

The `Makefile` I provided is a work around to just be able to compile descheduler on Windows. Many of the other make targets will still not work. None of the scripts in the `hack` directory will work on Windows.

I'd be happy to review a pull request that updates the [contributor guide](https://github.com/kubernetes-sigs/descheduler/blob/master/docs/contributor-guide.md) with any additional setup steps required when working with the Windows 10 Linux Subsystem.
